src/apps/filters/h3ToGeoBoundary.c

```diff
@@ -70,46 +64,68 @@ void doCell(H3Index h, int isKmlOut) {
 }
 
 int main(int argc, char *argv[]) {
-    // check command line args
-    if (argc > 5) {
-        fprintf(stderr, "usage: %s [outputMode kmlName kmlDesc]\n", argv[0]);
-        exit(1);
-    }
+    H3Index index = 0;
+    char userKmlName[BUFF_SIZE] = {0};
+    char userKmlDesc[BUFF_SIZE] = {0};
 
-    int isKmlOut = 0;
-    if (argc > 1) {
-        if (!sscanf(argv[1], "%d", &isKmlOut))
-            error("outputMode must be an integer");
+    Arg helpArg = {.names = {"-h", "--help"},
+                   .helpText = "Show this help message."};
+    Arg indexArg = {
+        .names = {"-i", "--index"},
+        .scanFormat = "%" PRIx64,
+        .valueName = "index",
+        .value = &index,
+        .helpText =
+            "Index, or not specified to read indexes from standard in."};
+    Arg kmlArg = {.names = {"-k", "--kml"},
+                  .helpText = "Print output in KML format."};
+    Arg kmlNameArg = {.names = {"-kn", "--kml-name"},
+                      .scanFormat = "%255c",  // BUFF_SIZE - 1
+                      .valueName = "name",
+                      .value = &userKmlName,
+                      .helpText = "Name of the KML file."};
+    Arg kmlDescArg = {.names = {"-kd", "--kml-description"},
+                      .scanFormat = "%255c",  // BUFF_SIZE - 1
+                      .valueName = "description",
+                      .value = &userKmlDesc,
+                      .helpText = "Description of the KML file."};
 
-        if (isKmlOut != 0 && isKmlOut != 1) error("outputMode must be 0 or 1");
-        char *defaultKmlName = "geo from H3";
-        char *defaultKmlDesc = "from h3ToGeo";
+    Arg *args[] = {&helpArg, &indexArg, &kmlArg, &kmlNameArg, &kmlDescArg};
 
-        char *kmlName = defaultKmlName;
-        char *kmlDesc = defaultKmlDesc;
+    if (parseArgs(argc, argv, 5, args, &helpArg,
+                  "Converts indexes to latitude/longitude cell boundaries in "
+                  "degrees")) {
+        return helpArg.found ? 0 : 1;
+    }
 
-        if (argc > 2) {
-            kmlName = argv[2];
-            if (argc > 3) kmlDesc = argv[3];
-        }
+    if (kmlArg.found) {
+        char *kmlName = "geo from H3";
+        if (kmlNameArg.found) kmlName = userKmlName;
+
+        char *kmlDesc = "from h3ToGeoBoundary";
```

Comment:
Same notes as above: `"H3 Geometry"` and `"Generated by h3ToGeoBoundary"` would be better IMO